pinc/Project.inc

```diff
@@ -18,9 +18,18 @@ class UTF8ConversionException extends Exception
 
 class ProjectException extends Exception
 {
+    // all project errors extend from here and in range 100-199
+    public function __construct($message, $code = 100)
+    {
+        parent::__construct($message, $code);
+    }
 }
 class NonexistentProjectException extends ProjectException
 {
+    public function __construct($message)
```

Comment:
I like the pattern you established of setting the code in the constructor definition and passing `$code` into the parent constructor. Let's continue that here and for the rest of them too.

Reply:
Addressing this comment is the only remaining thing for me -- after we resolve this I'm happy to get it in.

Reply:
Done this now, thanks.